elyra/pipeline/component_parameter.py

```diff
@@ -0,0 +1,804 @@
+        if not self.value:
+            validation_errors.append("Required annotation value was not specified.")
+
+        return validation_errors
+
+    def add_to_execution_object(self, runtime_processor: RuntimePipelineProcessor, execution_object: Any, **kwargs):
+        """Add KubernetesAnnotation instance to the execution object for the given runtime processor"""
+        runtime_processor.add_kubernetes_pod_annotation(instance=self, execution_object=execution_object, **kwargs)
+
+
+class KubernetesToleration(ElyraPropertyListItem):
+    """
+    Kubernetes tolerations to apply to the pod where the node is executed.
+    """
+
+    property_id = KUBERNETES_TOLERATIONS
+    generic = True
+    custom = True
+    _display_name = "Kubernetes Tolerations"
+    _json_data_type = "array"
+    _keys = ["key", "operator", "value", "effect"]
+    _ui_details_map = {
+        "key": {"display_name": "Key", "placeholder": "key", "json_type": "string", "required": False},
+        "operator": {"display_name": "Operator", "json_type": "string", "required": True},
+        "value": {"display_name": "Value", "placeholder": "value", "json_type": "string", "required": False},
+        "effect": {"display_name": "Effect", "placeholder": "NoSchedule", "json_type": "string", "required": False},
+    }
+
+    def __init__(self, key, operator, value, effect, **kwargs):
+        self.key = key
+        self.operator = operator
+        self.value = value
+        self.effect = effect
+
+    @classmethod
+    def create_instance(cls, prop_id: str, value: Optional[Any]) -> KubernetesToleration | None:
+        key, operator, value, effect = cls.unpack(value, "key", "operator", "value", "effect")
+        return KubernetesToleration(key=key, operator=operator, value=value, effect=effect)
+
+    @classmethod
+    def get_schema(cls) -> Dict[str, Any]:
+        """Build the JSON schema for an Elyra-owned component property"""
+        schema = super().get_schema()
+        op_enum = ["Equal", "Exists"]
+        schema["items"]["properties"]["operator"]["enum"] = op_enum
+        schema["items"]["properties"]["operator"]["default"] = op_enum[0]
+        schema["items"]["properties"]["effect"]["enum"] = ["", "NoExecute", "NoSchedule", "PreferNoSchedule"]
+        return schema
+
+    def get_all_validation_errors(self) -> List[str]:
+        """
+        Perform custom validation on an instance using the constraints documented in
+        https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
+        """
+        validation_errors = []
+
+        # Ensure the operator is valid
+        if self.operator not in ["Exists", "Equal"]:
+            validation_errors.append(
+                f"'{self.operator}' is not a valid operator: the value must be one of 'Exists' or 'Equal'."
+            )
+
+        if self.operator == "Equal" and (self.key is None or len(self.key) == 0):
```

Comment:
This can be cleaned up a bit using a Falsey...
```suggestion
        if self.operator == "Equal" and not self.key:
```